orc/src/main/java/org/apache/iceberg/orc/ORCSchemaUtil.java

```diff
@@ -308,7 +309,7 @@ private static TypeDescription buildOrcProjection(Integer fieldId, Type type, bo
           orcType = convert(fieldId, type, false);
         }
     }
-
+    orcType.setAttribute(ICEBERG_ID_ATTRIBUTE, fieldId.toString());
```

Comment:
Adding this is actually causing failures
```
org.apache.iceberg.data.orc.TestGenericReadProjection > testRenamedAddedField FAILED
    java.lang.IllegalArgumentException: No conversion of type LONG to self needed
        at org.apache.orc.impl.ConvertTreeReaderFactory.createAnyIntegerConvertTreeReader(ConvertTreeReaderFactory.java:1671)
        at org.apache.orc.impl.ConvertTreeReaderFactory.createConvertTreeReader(ConvertTreeReaderFactory.java:2124)
        at org.apache.orc.impl.TreeReaderFactory.createTreeReader(TreeReaderFactory.java:2331)
        at org.apache.orc.impl.TreeReaderFactory$StructTreeReader.<init>(TreeReaderFactory.java:1961)
        at org.apache.orc.impl.TreeReaderFactory.createTreeReader(TreeReaderFactory.java:2371)
        at org.apache.orc.impl.RecordReaderImpl.<init>(RecordReaderImpl.java:227)
        at org.apache.orc.impl.ReaderImpl.rows(ReaderImpl.java:752)
        at org.apache.iceberg.orc.OrcIterable.newOrcIterator(OrcIterable.java:80)
        at org.apache.iceberg.orc.OrcIterable.iterator(OrcIterable.java:65)
        at com.google.common.collect.Iterables.getOnlyElement(Iterables.java:254)
        at org.apache.iceberg.data.orc.TestGenericReadProjection.writeAndRead(TestGenericReadProjection.java:53)
```
And I vaguely remember we fixed a similar bug before in ORC

Reply:
It would be great to know what's going on here. Since this is just a projection schema and the reader is built with the Iceberg schema (that has required/optional), I don't think it is really a blocker. But setting a property here shouldn't cause ORC to fail, right?

Reply:
I'll file the necessary followups.